lms/djangoapps/program_enrollments/api/v1/tests/test_views.py

```diff
@@ -1401,77 +1407,30 @@ def test_403_if_not_enrolled_in_program(self):
-        course_enrollment = CourseEnrollmentFactory(
-            course_id=other_course_key,
-            user=self.student,
-        )
-
-        # add a second program course enrollment
-        ProgramCourseEnrollmentFactory.create(
-            program_enrollment=program_enrollment,
-            course_enrollment=course_enrollment,
-            course_key=other_course_key,
-            status='active',
-        )
-
-        # add a course over view for other_course_key
+    def _add_new_course_to_program(self, course_run_key, program):
+        """
+        Helper method to create another course, an overview for it,
+        add it to the program, and re-load the cache.
+        """
+        other_course_run = CourseRunFactory.create(key=text_type(course_run_key))
+        other_course = CourseFactory.create(course_runs=[other_course_run])
+        program['courses'].append(other_course)
+        self.set_program_in_catalog_cache(program['uuid'], program)
         CourseOverviewFactory.create(
-            id=other_course_key,
+            id=course_run_key,
             start=self.yesterday,
         )
 
-        self.client.login(username=self.student.username, password=self.password)
-        response = self.client.get(self.get_url(self.program_uuid))
-
-        self.assertEqual(status.HTTP_200_OK, response.status_code)
-        # we expect data associated with the last modified program enrollment
-        # with 'enrolled' status
-        self.assertEqual(
-            text_type(self.program_course_enrollment.course_key),
-            response.data['course_runs'][0]['course_run_id']
-        )
-
     def test_multiple_enrollments_all_enrolled(self):
-        # add a second program enrollment
-        program_enrollment = ProgramEnrollmentFactory.create(
-            program_uuid=self.program_uuid,
-            curriculum_uuid=self.other_curriculum_uuid,
-            user=self.student,
-        )
-
-        other_course_key_string = 'course-v1:edX+ToyX+Other_Course'
-        other_course_key = CourseKey.from_string(other_course_key_string)
+        other_course_key = CourseKey.from_string('course-v1:edX+ToyX+Other_Course')
+        self._add_new_course_to_program(other_course_key, self.program)
 
-        # add a second course enrollment
-        course_enrollment = CourseEnrollmentFactory(
+        # add a second course enrollment, which doesn't need a ProgramCourseEnrollment
+        # to be returned.
+        CourseEnrollmentFactory.create(
             course_id=other_course_key,
-            user=self.student
-        )
-
-        # add a second program course enrollment
-        ProgramCourseEnrollmentFactory.create(
-            program_enrollment=program_enrollment,
-            course_enrollment=course_enrollment,
-            course_key=other_course_key,
-            status='active',
+            user=self.student,
+            mode=CourseMode.VERIFIED,
         )
 
         # add a course over view for other_course_key
```

Comment:
is this not already done in `_add_new_course_to_program`?

Reply:
No, that function creates the fake course, run, and overview, but not the enrollment.

Reply:
I'm referring to the line below this (slightly outside of my comment range 😆  ) not above

Reply:
yes :)